pystac/extensions/projection.py

```diff
@@ -286,7 +286,8 @@ class ItemProjectionExtension(ProjectionExtension[pystac.Item]):
     """The :class:`~pystac.Item` properties, including extension properties."""
 
     def __init__(self, item: pystac.Item):
-        self.item = item
+        super().__init__(item)
+
```

Comment:
Actually, I think we should remove the `ProjectionExtension.__init__` method entirely so that it is consistent with the other extension implementations that work on Assets and Items. `ItemProjectionExtension` is the only sub-class that accepts an `item` argument to its init (`AssetProjectionExtension` has a different signature), so it seems like this may have been a mistake.

cc: @lossyrob

Reply:
+1

Reply:
Cool, can do!